novareach-solutions/movemate
# feat: Add comprehensive Swagger/OpenAPI documentation for all working API endpoints

## Description

This PR introduces comprehensive Swagger/OpenAPI documentation for our NestJS application endpoints. It includes detailed documentation for:

- Authentication flows (OTP request, verification, login)
- User management operations (CRUD)
- Agent management operations (signup, document handling, status management)
- Health check endpoints

Key improvements:
- Updated `pre-commit` to ensure the prettier format is included in the staged changes
- Using scalar for the UI representation of the API's (which I personally feel is muchhh better than swagger 🔥)
- Deleted all the test files that were included in the init of the application (❌)
- Added detailed request/response schemas with examples
- Documented all possible error scenarios (at least as much as I know, please update later accordingly 🥹)
- Added proper authentication requirements
- Included comprehensive examples for each endpoint (at least as much as I know, please update later accordingly 🥹)
- Added TypeScript DTOs with Swagger decorators
- Configured proper response types and status codes

## Type of change


## How Has This Been Tested?

1. Ran application locally and verified Scalar UI at `/docs`
2. Tested all example requests through Scalar UI
3. Verified authentication flows work as documented
4. Validated response schemas match API responses
5. Confirmed all error scenarios are properly documented

## Checklist:

- [x] All DTOs include proper @ApiProperty decorators
- [x] All endpoints include response examples
- [x] Error responses are properly documented
- [x] Authentication requirements are clearly specified
- [x] Swagger UI is accessible and functional

## Screenshots

**Landing page**
![image](https://github.com/user-attachments/assets/7c1e12b3-d48e-42e9-9424-e2d4aa1d8139)

One of the auth module **Request OTP** section:
![image](https://github.com/user-attachments/assets/635e7670-954d-4a9e-a1d9-9746a519d513)

## Additional context

‼️ **Keep in mind that this structure should now be followed by anyone who will be creating a new controller**. And also update the doc as the changes are made.

The Swagger documentation can be accessed at `/docs` after starting the application. The documentation includes:

- Proper JWT authentication setup
- Role-based access control information
- Example requests for each endpoint
- Detailed response schemas
- Error handling documentation
- Health check monitoring

This documentation will help both internal developers and API consumers understand and interact with our endpoints more effectively.

To test the documentation:
1. Start the application
2. Navigate to `/docs`
3. Use the "Try it out" feature to test endpoints
4. Verify response schemas match documentation
